src/handlers/multiSig/approvalChanged/approvalChanged.ts

```diff
@@ -66,7 +67,45 @@ export const handleMultiSigApprovalChanged: EventHandler = async (
       multiSigFromDB?.action?.type,
     );
 
-    if (notificationCategory && !motion.overallApprovalTimestamp.isZero()) {
+    let hasFullApproval = true;
+    let roleIndex = 0;
+
+    // Map over each permission.
+    while (
+      BigNumber.from(motion.requiredPermissions).gte(
+        BigNumber.from(1).shl(roleIndex),
+      )
+    ) {
+      // If the permission is required by this multi sig motion...
+      // (this logic is taken from the network, it's very cool I know)
+      if (
+        BigNumber.from(motion.requiredPermissions)
+          .and(BigNumber.from(1).shl(roleIndex))
+          .gt(BigNumber.from(0))
```

Comment:
Just wondering if it makes sense to extract `BigNumber.from(motion.requiredPermissions)` into a `const`?